dot/rpc/modules/system_test.go

```diff
@@ -1,4 +1,702 @@
 // Copyright 2021 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package modules
\ No newline at end of file
+package modules
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	apimocks "github.com/ChainSafe/gossamer/dot/rpc/modules/mocks"
+	testdata "github.com/ChainSafe/gossamer/dot/rpc/modules/test_data"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/multiformats/go-multiaddr"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSystemModule_BasicGetters(t *testing.T) {
+	var emptyMap map[string]interface{}
+	mockSystemAPI := new(apimocks.SystemAPI)
+	mockSystemAPI.On("ChainName").Return("polkadot", nil)
+	mockSystemAPI.On("SystemName").Return("kusama", nil)
+	mockSystemAPI.On("ChainType").Return("testChainType", nil)
+	mockSystemAPI.On("Properties").Return(emptyMap)
+	mockSystemAPI.On("SystemVersion").Return("1.2.1", nil)
+
+	mockNetworkAPI := new(apimocks.NetworkAPI)
+	mockNetworkAPI.On("Health").Return(common.Health{}, nil)
+	mockNetworkAPI.On("NetworkState").Return(common.NetworkState{}, nil)
+	mockNetworkAPI.On("Peers").Return([]common.PeerInfo{}, nil)
+
+	sm := NewSystemModule(mockNetworkAPI, mockSystemAPI, new(apimocks.CoreAPI), new(apimocks.StorageAPI), new(apimocks.TransactionStateAPI), new(apimocks.BlockAPI))
+
+	req := &EmptyRequest{}
+	var res string
+	var resMap interface{}
+	var sysHealthRes SystemHealthResponse
+	var networkStateRes SystemNetworkStateResponse
+	var sysPeerRes SystemPeersResponse
+
+	err := sm.Chain(nil, req, &res)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, "polkadot", res)
+
+	err = sm.Name(nil, req, &res)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, "kusama", res)
+
+	err = sm.ChainType(nil, req, &res)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, "testChainType", res)
+
+	err = sm.Properties(nil, req, &resMap)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, emptyMap, resMap)
+
+	err = sm.Version(nil, req, &res)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, "1.2.1", res)
+
+	err = sm.Health(nil, req, &sysHealthRes)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, SystemHealthResponse(common.Health{}), sysHealthRes)
+
+	err = sm.NetworkState(nil, req, &networkStateRes)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+
+	err = sm.Peers(nil, req, &sysPeerRes)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, SystemPeersResponse{}, sysPeerRes)
+}
+
+func TestSystemModule_TestNodeRoles(t *testing.T) {
```

Comment:
```suggestion
func TestSystemModule_NodeRoles(t *testing.T) {
```

Reply:
I did this initially but got errors since the integration test has a function of the same name

Reply:
I left as is for now, but maybe we let these default names be for the unit tests, then use a different naming convention for the integration tests